src/state/nftMarket/helpers.ts

```diff
@@ -59,7 +65,87 @@ export const getNftsMarketData = async (collectionAddress: string): Promise<Nfts
     )
     return res.collection.NFTs
   } catch (error) {
-    console.error('Failed to fetch NFTs', error)
+    console.error('Failed to fetch NFTs from collection', error)
+    return []
+  }
+}
+
+export const getNftsMarketData = async (where = {}): Promise<NftSubgraphEntity[]> => {
+  try {
+    const res = await request(
+      GRAPH_API_NFTMARKET,
+      gql`
+        query getNftsMarketData($where: NFT_filter) {
+          nfts(where: $where) {
+            tokenId
+            metadataUrl
+            transactionHistory {
+              id
+              block
+              timestamp
+              askPrice
+              netPrice
+              buyer {
+                id
+              }
+              seller {
+                id
+              }
+              withBNB
+            }
+          }
+        }
+      `,
+      { where },
+    )
+
+    return res.nfts
+  } catch (error) {
+    console.error('Failed to fetch NFTs market data', error)
     return []
   }
 }
+
+export const fetchWalletNftIds = async (account: string): Promise<string[]> => {
```

Comment:
Reworked version of `fetchWalletNfts` - when `/collectibles` is deleted, I'll migrate everything to use this function